src/color.rs

```diff
@@ -20,6 +38,8 @@ impl Color {
     }
 
     pub fn blend(&self, other: &Color, ratio: f32) -> Color {
+        validate_color(self);
+        validate_color(other);
```

Comment:
It shouldn't be necessary to validate on blend if all colors are validated when constructed, no?